django/contrib/postgres/fields/array.py

```diff
@@ -262,7 +262,7 @@ def __init__(self, index, base_field, *args, **kwargs):
 
     def as_sql(self, compiler, connection):
         lhs, params = compiler.compile(self.lhs)
-        return '%s[%s]' % (lhs, self.index), params
+        return '%s[%%s]' % lhs, [self.index] + params
```

Comment:
Shouldn't that have params and index swapped? I fear we also might have to fix this in the existing released patches?

Reply:
If the answer to my previous question is yes the we also need a new test for that, apparently we nowhere tested with extra params :D

Reply:
Currently both can be used only with field name, i.e. `params` are always an empty array `[]` so it's not an issue.

Reply:
Swapped.